External/CoordGen/CoordGen.h

```diff
@@ -156,19 +210,30 @@ unsigned int addCoords(T& mol, const CoordGenParams* params = nullptr) {
   }
 
   minimizer.initialize(min_mol);
-  minimizer.runGenerateCoordinates();
+  if (!params->minimizeOnly) {
+    minimizer.runGenerateCoordinates();
+  } else {
+    CoordgenFragmenter::splitIntoFragments(min_mol);
+    minimizer.m_minimizer.minimizeMolecule(min_mol);
+  }
   auto conf = new Conformer(mol.getNumAtoms());
   for (size_t i = 0; i < mol.getNumAtoms(); ++i) {
-    conf->setAtomPos(
-        i, RDGeom::Point3D(ats[i]->coordinates.x() / scaleFactor,
-                           ats[i]->coordinates.y() / scaleFactor, 0.0));
+    auto coords = RDGeom::Point3D(ats[i]->coordinates.x() / scaleFactor,
+                                  ats[i]->coordinates.y() / scaleFactor, 0.0);
+    if (params->minimizeOnly) {
+      // readjust the average bond length from 1 (coordgen's target)
+      // to whatever we started with
+      coords *= avgBondLen;
```

Comment:
avgBondLen could be shoved into scaleFactor.

Reply:
indeed, but I think having it explicit is useful